packages/@headlessui-react/src/hooks/document-overflow/adjust-scrollbar-padding.ts

```diff
@@ -0,0 +1,32 @@
+import { ScrollLockRequest } from './request'
+
+export function adjustScrollbarPadding(
+  req: ScrollLockRequest,
+  next: (req: ScrollLockRequest) => void
+) {
+  let { doc } = req
+
+  if (!req.isLocked) {
+    next(req)
+    doc.documentElement.style.paddingRight = req.ctx.oldPaddingRight
+    return
+  }
+
+  let documentElement = doc.documentElement
+
+  // Record the scrollbar width before we change the style
+  let ownerWindow = doc.defaultView ?? window
+  let scrollbarWidthBefore = ownerWindow.innerWidth - documentElement.clientWidth
+
+  req.ctx.oldPaddingRight = documentElement.style.paddingRight
+
+  // Update the overflow style of the document itself
```

Comment:
Nit: drop this since this comment is technically wrong from the `adjustScrollbarPadding`'s perspective since it depends on what other middleware is defined (and what order). (I think?)